ompi/request/req_wait.c

```diff
@@ -108,6 +110,8 @@ int ompi_request_default_wait_any(size_t count,
             completed = i;
             *index = i;
             goto after_sync_wait;
+        } else {
+            sync_sets++;
```

Comment:
at the end of this loop sync_sets = completed - 1 - num_requests_null_inactive (at after_sync_wait:) ;)

In fact we don't need to count here because if at the end *index is not equal to completed it means that one of the originally not-completed requests have been completed during the wait.